test/utils/runners.go

```diff
@@ -672,3 +674,157 @@ func DoCleanupNode(client clientset.Interface, nodeName string, strategy Prepare
 	}
 	return fmt.Errorf("To many conflicts when trying to cleanup Node %v", nodeName)
 }
+
+type TestPodCreateStrategy func(client clientset.Interface, namespace string, podCount int) error
+
+type TestPodCreatorConfig map[string]map[int]TestPodCreateStrategy
```

Comment:
I'm not a fan of such complicated maps.
How about splitting this into two:
```
PodCreatorConfig (probably better name) struct {
  namespace string
  numPods int
  creator func(...)
}
```

and then:
```
type ConfigList []Config   (with better names)
```